src/versionbits_impl.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2016-2022 The Bitcoin Core developers
```

Comment:
In commit 49afd714b40115b15ea0ef146b48e9d7a403a596

Can you give some more motivation for splitting this out into its own header, especially since it seems to be introducing a new circular dependency? It also not immediately clear to me why a file called implementation only holds abstract declarations.  From what I understand this is supposed to hold declarations that are only relevant for external use by tests.